### PR TITLE
Drop the workaround to allow paths with non ascii chars

### PR DESCRIPTION
**Pull request by prusse-martin**
In recent hdf5 versions the old problem with non ascii paths apper to be fixed. A test to check this support have been added.

ASIM-5880

Related PR (commit) in alfasim: https://github.com/ESSS/alfasim/pull/731 (https://github.com/ESSS/alfasim/pull/731/commits/98f610fed968b8cc7b3dec45ce7702058575b5ed)

### CODE REVIEW

File: tests/results/test_aggregator.py [hunk 1]
```diff
@@ -313,6 +315,22 @@ def test_read_trends_data_empty_arrays_when_no_time_set_info(results: Results) -
     assert all([len(trend_data) == 0 for trend_data in trends.values()])
 
 
+def test_fancy_path(datadir: Path, request: FixtureRequest):
+    """
+    The setup is based on `results` fixture (from `conftest.py`).
```
Comment (nicoddemus) [inline tests/results/test_aggregator.py:320]:
```suggestion
    Ensure the HDF5 version we are using can work with paths containing non-ascii characters (ASIM-5880).
    
    The setup is based on `results` fixture (from `conftest.py`).
```

File: tests/results/test_aggregator.py [hunk 2]
```diff
@@ -9,6 +10,7 @@
 import h5py
 import numpy
 import pytest
+from _pytest.fixtures import FixtureRequest
```
Comment (nicoddemus) [inline tests/results/test_aggregator.py:13]:
```suggestion
from pytest import FixtureRequest
```

File: environment.devenv.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ dependencies:
   - boltons
   - click
   - h5py
+  - hdf5 >= 1.10
```
Comment (nicoddemus) [inline environment.devenv.yml:10]:
```suggestion
  # Supports non-ascii paths (ASIM-5880). 
  - hdf5 >= 1.10
```

**Review by nicoddemus — APPROVED** [review]

File: tests/results/test_aggregator.py (hunk 2 above)
Comment (prusse-martin) [reply tests/results/test_aggregator.py:13]: *scream in despair* Why did you fail me auto import!

**Review by ro-oliveira95 — APPROVED** [review]

**Review by BeneBr — APPROVED** [review]

**Pull request merged**